src/auth/strategies/local.strategy.ts

```diff
@@ -0,0 +1,22 @@
+import { Strategy } from 'passport-local';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { AuthService } from '../auth.service';
+import { AuthDto } from '../dto';
+import { User } from '@prisma/client';
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private authService: AuthService) {
+    super({ usernameField: 'email' });
+  }
+
+  async validate(email: string, password: string): Promise<User> {
+    const data = { email, password };
+    const user = await this.authService.validateUser(data);
+    if (!user) {
+      throw new UnauthorizedException();
+    }
+    return user;
```

Comment:
I like to add new space before code that opens `{}` like e.g. `if () {...}` and before `return`, like this:
```suggestion
    const user = await this.authService.validateUser(data);
    
    if (!user) {
      throw new UnauthorizedException();
    }
    
    return user;
```

but I know developers have their own preferences so I'm not forcing you to change it 🙂